src/main/webapp/upcoming-events.jsp

```diff
@@ -57,7 +57,8 @@
 
         try {
             upcomingEvents = service.getUpcomingEvents(username);
-            upcomingEvents = upcomingEvents.subList(0, numberOfActivities);
+            upcomingEvents = upcomingEvents.subList(0, numberOfActivities < 0 ? 0 : (numberOfActivities >
```

Comment:
`toIndex` is exclusive, dus deze code klopt niet (want je moet geen -1 doen).

Verder vind ik `Math.min(numberOfActivities, upcomingEvents.size())` ook wel prima, en is die `< 0` check een beetje overbodig als je het mij vraagt.